### ISSUE TITLE
Extend GSI to read netcdf global_berror files

### ISSUE THREAD

**Issue by RussTreadon-NOAA** [NOAA-EMC/GSI#808]
JEDI can read both binary and netcdf format global_berror files.  This issue is opened to document that changes needed to allow the GSI to read netcdf format global_berror files.

**Comment by RussTreadon-NOAA** [comment]
Work for this issue will be done in [`feature/netcdf_berror`](https://github.com/RussTreadon-NOAA/GSI/tree/feature/netcdf_berror)

**Comment by RussTreadon-NOAA** [comment]
Copy the following files from a working copy of GDASApp `develop sorc/gsibec/src/gsibec/gsi` at [`c8ac58d`](https://github.com/GEOS-ESM/GSIbec/tree/c8ac58d9b43eb8f890e565d12c88f1b0579c9ccd)
 - `m_berror_stats.f90`
 - `m_nc_berror.f90`
to GSI `src/gsi`. 

Several modifications needed to get `m_berror_stats.f90` to work in GSI.   Changes committed to `feature/netcdf_berror` at 9456a0b.

**Comment by RussTreadon-NOAA** [comment]
Add `read_iasing.f90` to `src/gsi/gsi_files.cmake`.  [9ada88e](https://github.com/NOAA-EMC/GSI/commit/9ada88e8b080c37a5146092e6a0a8dfc45b6654b) added file `read_iasing.f90` to `develop`.

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 ctests**
Install `feature/netcdf_berror` at 5f348de3 on Dogwood.  Make the following modifications in working copy of `feature/netcdf_berror`
- add logic and scripting to `regression/global_4denvar.sh` to use a netcdf `berror_stats` file for _updat_ and a binary `berror_stats` file for _contrl_
- add netcdf global_berror files to `fix/`

Run ctests with the following results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/berror/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  847.00 sec
2/6 Test #6: global_enkf ......................   Passed  1391.05 sec
3/6 Test #2: rtma .............................   Passed  1448.58 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  1512.89 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1579.10 sec
6/6 Test #1: global_4denvar ...................   Passed  2522.57 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 2522.72 sec
```

Confirm that _updat_ global_4denvar runs used netcdf `berror_stats` file.  The _contrl_ runs used the binary `berror_stats` file.  Analysis results are identical when comparing _updat_ and _contrl_.

**Comment by RussTreadon-NOAA** [comment]
**Resolution tests**

Install `feature/netcdf_berror` at c484d4c8 on Dogwood.  

Set up stand-alone rungsi script to run 20241126 00Z gdas using binary and netcdf `global_berror` files.  Configure `gsi.x` to run in 3dvar mode without FGAT.  Read in operational resolution backgrounds and observations with analysis performed on the following analysis grids using the indicated binary and netcdf `global_berror` files
- C768, [1536, 768], `global_berror.l127y770.[f77,nc]`
- C384, [768, 384], `global_berror.l127y386.[f77,nc]`
- C192, [384, 192], `global_berror.l127y194.[f77,nc]`
- C96, [192, 96], `global_berror.l127y98.[f77,nc]`

The analysis increment file, `siginc.nc` is bitwise identical between or the binary `.f77` and netcdf `.nc` `global_berror` runs.  The `gsi.x` wall time and task 0 maximum resident set size are comparable when using the binary or netcdf `global_berror` file.

**Comment by RussTreadon-NOAA** [comment]
Open g-w issue [#3128](https://github.com/NOAA-EMC/global-workflow/issues/3128) to add netcdf `global_berror` files to staged GSI fix directory.

Open GSI-fix PR [#25](https://github.com/NOAA-EMC/GSI-fix/pull/25) to add netcdf `global_berror` files to `gsi_binary_files.cmake`.
